packages/react-router/__tests__/matchPath-test.tsx

```diff
@@ -298,38 +298,24 @@ describe("matchPath *", () => {
   });
 });
 
-describe("matchPath warnings", () => {
-  let consoleWarn: jest.SpyInstance<void, any>;
-  beforeEach(() => {
-    consoleWarn = jest.spyOn(console, "warn").mockImplementation();
-  });
-
-  afterEach(() => {
-    consoleWarn.mockRestore();
-  });
-
-  describe("when the pattern has a trailing *", () => {
-    it("issues a warning and matches the remaining portion of the pathname", () => {
-      expect(matchPath("/files*", "/files/mj.jpg")).toMatchObject({
-        params: { "*": "mj.jpg" },
-        pathname: "/files/mj.jpg",
-        pathnameBase: "/files",
-      });
-      expect(consoleWarn).toHaveBeenCalledTimes(1);
+describe("when the pattern has a trailing *", () => {
+  it("matches the remaining portion of the pathname", () => {
+    expect(matchPath("/files*", "/files/mj.jpg")).toMatchObject({
+      params: { "*": "/mj.jpg" },
```

Comment:
Previously trimmed leading slashes are now included since we support partial splat matching